pkg/cli/cmds/etcd_snapshot.go

```diff
@@ -0,0 +1,38 @@
+package cmds
+
+import (
+	"github.com/rancher/k3s/pkg/version"
+	"github.com/urfave/cli"
+)
+
+const EtcdSnapshotCommand = "etcd-snapshot"
+
+func NewEtcdSnapshotCommand(action func(*cli.Context) error) cli.Command {
+	return cli.Command{
+		Name:            EtcdSnapshotCommand,
+		Usage:           "Run an Etcd snapshot",
+		SkipFlagParsing: false,
+		SkipArgReorder:  true,
+		Action:          action,
+		Flags: []cli.Flag{
+			DebugFlag,
+			LogFile,
+			AlsoLogToStderr,
+			cli.StringFlag{
+				Name:        "data-dir,d",
+				Usage:       "(data) Folder to hold state default /var/lib/rancher/" + version.Program + " or ${HOME}/.rancher/" + version.Program + " if not root",
+				Destination: &ServerConfig.DataDir,
+			},
+			&cli.StringFlag{
+				Name:        "snapshot-name",
```

Comment:
I like the idea you had here - `k3s server --etcd-snapshot-name=x` becomes `k3s etcd-snapshot --name=x` but changing the flag names means that it breaks setting these things via the config yaml. They need to be the same across all commands.

For example, I might have `etcd-snapshot-dir: /opt/etcd-snapshots/` in /etc/k3s/config.yaml, and would expect `k3s etcd-snapshot` to honor that setting, but it will not because the flag on this command is just `dir`.

While you're at it, I think we should probably also expose the --etcd-snapshot-retention flag here.